test/integration/change-streams/change_stream.test.ts

```diff
@@ -1816,6 +1829,78 @@ describe('Change Streams', function () {
       });
     });
   });
+
+  describe("NODE-4763 - doesn't produce duplicates after resume", function () {
+    const resumableError = { code: 6, message: 'host unreachable' };
+
+    beforeEach(async function () {
+      await client.db('admin').command({
+        configureFailPoint: is4_2Server(this.configuration.version)
+          ? 'failCommand'
+          : 'failGetMoreAfterCursorCheckout',
+        mode: { skip: 1 },
+        data: {
+          failCommands: ['getMore'],
+          errorCode: resumableError.code,
+          errmsg: resumableError.message
+        }
+      } as FailPoint);
+    });
+
+    afterEach(async function () {
+      await client.db('admin').command({
+        configureFailPoint: is4_2Server(this.configuration.version)
+          ? 'failCommand'
+          : 'failGetMoreAfterCursorCheckout',
+        mode: 'off'
+      } as FailPoint);
+    });
+
+    describe('when using iterator form', function () {
+      it('#next', { requires: { topology: 'replicaset' } }, async function test() {
+        await initIteratorMode(changeStream);
```

Comment:
When I ran this test locally, it still passes if I comment out the failpoint configuration in line 1837.

For these tests, we need something that actually asserts that the change stream was resumed.  In above tests, we've used [command monitoring](https://www.mongodb.com/docs/drivers/node/current/monitoring-and-logging/monitoring/) and counting the number of aggregate events we expect (in these tests, we'd expect 2: once for the initial change stream creation, another for the resume attempt).